### PR TITLE
Update typehints for all the transforms

### PR DESCRIPTION
**Pull request by Nic-Ma**
### Description
This PR mainly include:
1. Fixed wrong `typehints` for all the args that support `ensure_tuple_rep`.
2. Made `Resized` and `RandResized` transform to support different `align_corners` for different keys.
3.  Removed wrong doc-string of `spatial_size` for `Resized` transform.
4. Gave `spatial_size` of `RandAffined` the default value=None according to its behavior.
5. Set typehints for many missing places.
6. Added missing doc-strings for Affine args in `RandAffined`, etc.

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/spatial/array.py [hunk 1]
```diff
@@ -1035,16 +1066,13 @@ def __init__(
...
-            scale_params (list of floats):
-                a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Defaults to no scaling.
-            spatial_size (list or tuple of int): output image spatial size.
+            rotate_params: a rotation angle in radians, a scalar for 2D image, a tuple of 3 floats for 3D.
+                Defaults to no rotation.
+            shear_params: a tuple of 2 floats for 2D, a tuple of 6 floats for 3D. Defaults to no shearing.
+            translate_params: a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Translation is in
+                pixel/voxel relative to the center of the input image. Defaults to no translation.
+            scale_params: a tuple of 2 floats for 2D, a tuple of 3 floats for 3D. Defaults to no scaling.
+            spatial_size: output image spatial size.
```
Comment (wyli) [inline monai/transforms/spatial/array.py:1075]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/array.py [hunk 2]
```diff
@@ -1121,7 +1149,24 @@ def __init__(
...
+                N-th parameter.
+            translate_range : translate_range[0] with be used to generate the 1st shift parameter from
+                `uniform[-translate_range[0], translate_range[0])`. Similarly, `translate_range[1]`
+                to `translate_range[N]` controls the range of the uniform distribution used to generate
+                the 2nd to N-th parameter.
+            scale_range: scaling_range[0] with be used to generate the 1st scaling factor from
+                `uniform[-scale_range[0], scale_range[0]) + 1.0`. Similarly, `scale_range[1]` to
+                `scale_range[N]` controls the range of the uniform distribution used to generate the 2nd to
+                N-th parameter.
+            spatial_size: output image spatial size.
```
Comment (wyli) [inline monai/transforms/spatial/array.py:1167]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/array.py [hunk 3]
```diff
@@ -1206,28 +1251,40 @@ class Rand2DElastic(Randomizable, Transform):
 
     def __init__(
         self,
-        spacing,
-        magnitude_range,
+        spacing: Sequence[float],
```
Comment (wyli) [inline monai/transforms/spatial/array.py:1254]: `spacing: Union[Tuple[float, float], float]`

File: monai/transforms/spatial/array.py [hunk 4]
```diff
@@ -1206,28 +1251,40 @@ class Rand2DElastic(Randomizable, Transform):
...
+                `uniform[-shear_range[0], shear_range[0])`. Similarly, `shear_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            translate_range : translate_range[0] with be used to generate the 1st shift parameter from
+                `uniform[-translate_range[0], translate_range[0])`. Similarly, `translate_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            scale_range: scaling_range[0] with be used to generate the 1st scaling factor from
+                `uniform[-scale_range[0], scale_range[0]) + 1.0`. Similarly, `scale_range[1]` controls
+                the range of the uniform distribution used to generate the 2nd parameter.
+            spatial_size: specifying output image spatial size [h, w].
+                if `spatial_size` and `self.spatial_size` are not defined, or smaller than 1,
```
Comment (wyli) [inline monai/transforms/spatial/array.py:1286]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/array.py [hunk 5]
```diff
@@ -1316,29 +1373,44 @@ class Rand3DElastic(Randomizable, Transform):
...
+                `uniform[-shear_range[0], shear_range[0])`. Similarly, `shear_range[1]` and `shear_range[2]`
+                controls the range of the uniform distribution used to generate the 2nd and 3rd parameters.
+            translate_range : translate_range[0] with be used to generate the 1st shift parameter from
+                `uniform[-translate_range[0], translate_range[0])`. Similarly, `translate_range[1]` and
+                `translate_range[2]` controls the range of the uniform distribution used to generate
+                the 2nd and 3rd parameters.
+            scale_range: scaling_range[0] with be used to generate the 1st scaling factor from
+                `uniform[-scale_range[0], scale_range[0]) + 1.0`. Similarly, `scale_range[1]` and `scale_range[2]`
+                controls the range of the uniform distribution used to generate the 2nd and 3rd parameters.
+            spatial_size: specifying output image spatial size [h, w, d].
```
Comment (wyli) [inline monai/transforms/spatial/array.py:1411]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/utility/dictionary.py [hunk 6]
```diff
@@ -347,14 +347,14 @@ class CopyItemsd(MapTransform):
 
     """
 
-    def __init__(self, keys: KeysCollection, times: int, names):
+    def __init__(self, keys: KeysCollection, times: int, names: Union[Sequence[str], str]):
```
Comment (wyli) [inline monai/transforms/utility/dictionary.py:350]: `: Union[Sequence[str], str]`-> `KeysCollection`

File: monai/transforms/spatial/dictionary.py [hunk 7]
```diff
@@ -267,13 +269,12 @@ class Resized(MapTransform):
     Args:
         keys: keys of the corresponding items to be transformed.
             See also: :py:class:`monai.transforms.compose.MapTransform`
-        spatial_size (tuple or list): expected shape of spatial dimensions after resize operation.
-            If `spatial_size` are not defined, or smaller than 1, the transform will use the spatial size of `img`.
+        spatial_size: expected shape of spatial dimensions after resize operation.
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:272]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/dictionary.py [hunk 8]
```diff
@@ -404,15 +420,26 @@ def __init__(
         """
         Args:
             keys: keys of the corresponding items to be transformed.
-            spatial_size (2 ints): specifying output image spatial size [h, w].
+            spacing: distance in between the control points.
+            magnitude_range: 2 int numbers, the random offsets will be generated from
+                ``uniform[magnitude[0], magnitude[1])``.
+            spatial_size: specifying output image spatial size [h, w].
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:426]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/dictionary.py [hunk 9]
```diff
@@ -503,16 +530,29 @@ def __init__(
         """
         Args:
             keys: keys of the corresponding items to be transformed.
-            spatial_size (3 ints): specifying output image spatial size [h, w, d].
+            sigma_range: a Gaussian kernel with standard deviation sampled from
+                ``uniform[sigma_range[0], sigma_range[1])`` will be used to smooth the random offset grid.
+            magnitude_range: the random offsets on the grid will be generated from
+                ``uniform[magnitude[0], magnitude[1])``.
+            spatial_size: specifying output image spatial size [h, w, d].
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:537]: If the components of the `spatial_size` are non-positive values, the transform will use the corresponding components of `img` size. For example, `spatial_size=(32, -1)` will be adapted to `(32, 64)` if the second spatial dimension size of `img` is `64`.

File: monai/transforms/spatial/dictionary.py [hunk 10]
```diff
@@ -701,21 +746,22 @@ class RandRotated(Randomizable, MapTransform):
...
+            It also can be a sequence of bool, each element corresponds to a key in ``keys``.
     """
 
     def __init__(
         self,
         keys: KeysCollection,
-        range_x=0.0,
-        range_y=0.0,
-        range_z=0.0,
+        range_x: Union[Sequence[float], float] = 0.0,
```
Comment (wyli) [inline monai/transforms/spatial/dictionary.py:757]: ?

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review!
I have updated the PR according to your comments.
Thanks.

**Pull request merged**